cg/store/crud/read.py

```diff
@@ -1528,3 +1545,67 @@ def get_case_ids_for_samples(self, sample_ids: list[int]) -> list[str]:
         for sample_id in sample_ids:
             case_ids.extend(self.get_case_ids_with_sample(sample_id))
         return list(set(case_ids))
+
+    def _get_related_dna_samples_from_sample_within_collaborators(
+        self, sample_internal_id: str, collaborators: set[Customer]
+    ) -> list[Sample]:
+        """Returns a DNA sample with the same subject_id, tumour status and within the collaborators of an RNA sample."""
+        sample_application_version_query: Query = self._get_join_sample_application_version_query()
+        sample_application_version_query: Query = apply_application_filter(
+            applications=sample_application_version_query,
+            filter_functions=[ApplicationFilter.IS_DNA],
+        )
+
+        sample: Sample = self.get_sample_by_internal_id(internal_id=sample_internal_id)
+        customer_entry_ids: list[str] = [customer.id for customer in collaborators]
+        sample_filter_functions: list[SampleFilter] = [
+            SampleFilter.BY_SUBJECT_ID,
+            SampleFilter.BY_TUMOUR,
+            SampleFilter.BY_CUSTOMER_ENTRY_IDS,
+        ]
+
+        sample_application_version_query: Query = apply_sample_filter(
+            samples=sample_application_version_query,
+            subject_id=sample.subject_id,
+            is_tumour=sample.is_tumour,
+            customer_entry_ids=customer_entry_ids,
+            filter_functions=sample_filter_functions,
+        )
+
+        return sample_application_version_query.all()
+
+    def _get_dna_cases_from_sample_within_collaborators(
+        self, sample_internal_id: str, collaborators: set[Customer]
+    ) -> list[Case]:
+        """Return a list of DNA cases linked to DNA sample with a list of customers in a collaboration."""
+
+        case_samples: Query = self._get_join_case_sample_query()
+        case_samples: Query = apply_case_sample_filter(
+            case_samples=case_samples,
+            sample_internal_id=sample_internal_id,
+            filter_functions=[CaseSampleFilter.CASES_WITH_SAMPLE_BY_INTERNAL_ID],
+        )
+
+        workflows = [
+            Workflow.MIP_DNA,
+            Workflow.BALSAMIC,
+            Workflow.BALSAMIC_UMI,
+        ]
```

Comment:
Can these be passed to this method instead?

Reply:
Then should this method be more general? Named something like  "_get_cases_from_sample_within_collaborators_and_workflows"?

Reply:
Yes, and they don't need to be private methods (i.e. with the "_") These are queries that could be used elsewhere.

Reply:
Changed!